src/main/java/com/forgeessentials/api/UserIdent.java

```diff
@@ -214,39 +271,111 @@ public static synchronized UserIdent get(EntityPlayerMP player)
         else
         {
             String name = player.getCommandSenderName();
-            if (name != null && !name.equals(ident.username))
+
+            String newname = name + "_" + ident.uuid.hashCode();
+            if (name != null && !name.equals(ident.username) && !newname.equals(ident.username))
             {
-                byUsername.remove(ident.username);
+                if (byUsername.containsKey(name.toLowerCase()))
```

Comment:
And since fake players use the same api as players, when a fake player changes their name, it can overwrite an existing name.  This is usually not a problem except for mods like https://github.com/MrTJP/ProjectRed/blob/375f4e4cafb8c66eb33b99213046fec5fd723426/src/mrtjp/projectred/expansion/TileBlockPlacer.scala who keep changing their identity.